core/controllers/question.py

```diff
@@ -28,7 +28,12 @@ class QuestionsBatchHandler(base.BaseHandler):
     def get(self):
         """Handles GET requests."""
         collection_id = self.request.get('collection_id')
-        skill_ids = json.loads(self.request.get('stringified_skill_ids'))
+        if not collection_id:
+            raise self.PageNotFoundException
+        if self.request.get('stringified_skill_ids'):
+            skill_ids = json.loads(self.request.get('stringified_skill_ids'))
```

Comment:
Let's just do self.PageNotFoundException here too. Better to force the caller to get it right.

Reply:
Updated.